### PR TITLE
fix(Tumblr): fix issues by rewriting presence

### LINKED ISSUE #7950: Tumblr | tumblr.com - Not working
### Steps to Reproduce

1 - Add presence;
2 - Access the website, on any page;
3 - Go to Discord and observe.

### Expected Behavior

The presence would be appearing

### Actual Behavior

Presence does not appear

### Page URL

https:// [userpage] .tumblr.com

### Screenshots

image

### Operating System

Windows 10

### Browser and Version

Opera GX LVL 5 (core: 105.0.4970.76) (64 Bits)

### Additional Comments

I believe it is not working because the last presence update was in 2019 (#578), when it was added.

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

- Resolves #7950
- Fix issues;
- Shorten document.location.pathname/hostname/href to variables;
- Change if/else if/else to switch
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/T/Tumblr/presence.ts [hunk 1]
```diff
@@ -3,105 +3,178 @@ const presence = new Presence({
...
+					break;
+				}
+				case pathname === "/": {
+					presenceData.details = "Viewing the home page";
+					break;
+				}
+				case document.querySelector<HTMLMetaElement>('[property="og:type"]')
+					?.content === "profile": {
+					presenceData.details = `Viewing user: ${
+						document.querySelector(".vfPi2").textContent
```
Comment (theusaf) [inline websites/T/Tumblr/presence.ts:73]: Try non-generated selectors if possible.
```suggestion
						document.querySelector("header h1").textContent
```

File: websites/T/Tumblr/presence.ts [hunk 2]
```diff
@@ -3,105 +3,178 @@ const presence = new Presence({
...
+					}`;
+					presenceData.state =
+						document.querySelector(".MLzKD.PqK1Y")?.textContent; // Tab;
```
Comment (theusaf) [inline websites/T/Tumblr/presence.ts:76]:
```suggestion
						document.querySelector("header + div a[class*=' ']")?.textContent; // Tab;
```
```suggestion
[...document.querySelectorAll("header + div a")].find(e => e.classList.length > 1)?.textContent
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/T/Tumblr/presence.ts (hunk 1 above)
Comment (DarkVillager) [reply websites/T/Tumblr/presence.ts:73]: > Try non-generated selectors if possible.

not possible

**Review by Bas950 — APPROVED** [review]
Except for the points of Usaf, LGTM. 
Don't get why peeps don't split things into multiple files tho (I would personally split your hostname switch to link to functions in a different file so its more clear, `case <hostname>: <functionname>(presencedata)`)

File: websites/T/Tumblr/presence.ts (hunk 1 above)
Comment (theusaf) [reply websites/T/Tumblr/presence.ts:73]: Do the suggested selectors I provided not work?

**Comment by DarkVillager** [comment]
> Don't get why peeps don't split things into multiple files tho (I would personally split your hostname switch to link to functions in a different file so its more clear, `case <hostname>: <functionname>(presencedata)`)
For websites with small help pages its not worth it imo, but for sites that have big secondary pages, I'll do it it in the future.

**Review by theusaf — APPROVED** [review]

**Pull request merged**